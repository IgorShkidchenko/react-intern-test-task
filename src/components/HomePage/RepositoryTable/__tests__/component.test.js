import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import RepositoryRowComponent from "../../RepositoryRow/component";
import RepositoryTableComponent from "../component";

configure({ adapter: new Adapter() });

describe("<RepositoryTableComponent />", () => {
  it("renders correctly", () => {
    const rows = [
      <RepositoryRowComponent
        repository={{ name: "name", stargazers_count: 1 }}
        handleClick={() => {}}
        key="name"
      />,
      <RepositoryRowComponent
        repository={{ name: "name2", stargazers_count: 2 }}
        handleClick={() => {}}
        key="name2"
      />,
    ];
    const wrapper = shallow(<RepositoryTableComponent rows={rows} />);
    expect(wrapper).toMatchSnapshot();
  });
});

