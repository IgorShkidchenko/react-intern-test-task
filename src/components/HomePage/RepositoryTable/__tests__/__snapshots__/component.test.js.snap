// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryTableComponent /> renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Repository Name
      </th>
      <th>
        Stars
      </th>
    </tr>
  </thead>
  <tbody>
    <RepotisoryRowComponent
      handleClick={[Function]}
      key="name"
      repository={
        Object {
          "name": "name",
          "stargazers_count": 1,
        }
      }
    />
    <RepotisoryRowComponent
      handleClick={[Function]}
      key="name2"
      repository={
        Object {
          "name": "name2",
          "stargazers_count": 2,
        }
      }
    />
  </tbody>
</table>
`;
